client/state/data-layer/wpcom/sites/comments/replies/new/index.js

```diff
@@ -4,20 +4,29 @@
 import { COMMENTS_WRITE } from 'state/action-types';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import {
+	hasParentCommentId,
 	dispatchNewCommentRequest,
 	updatePlaceholderComment,
 	handleWriteCommentFailure,
 } from 'state/data-layer/wpcom/sites/utils';
 
 export const writePostComment = ( { dispatch }, action ) => {
-	if ( ! action.parentCommentId ) {
+	if ( ! hasParentCommentId( action ) ) {
 		return;
 	}
 
 	const path = `/sites/${ action.siteId }/comments/${ action.parentCommentId }/replies/new`;
 	dispatchNewCommentRequest( dispatch, action, path );
 };
 
+const onSuccess = ( { dispatch }, action, ...args ) => ! hasParentCommentId( action )
+	? updatePlaceholderComment( { dispatch }, action, ...args )
+	: null;
+
+const onFailure = ( { dispatch }, action, ...args ) => ! hasParentCommentId( action )
+	? handleWriteCommentFailure( { dispatch }, action, ...args )
+	: null;
+
 export default {
-	[ COMMENTS_WRITE ]: [ dispatchRequest( writePostComment, updatePlaceholderComment, handleWriteCommentFailure ) ]
+	[ COMMENTS_WRITE ]: [ dispatchRequest( writePostComment, onSuccess, onFailure ) ]
```

Comment:
Unless we have a way of inspecting the action payload before trigging the  related handlers, I'm thinking that adding another action type like `COMMENTS_REPLY_WRITE` would be clearer. If we do go forward with this, I think it would be worthwhile to at least add a quick comment on the other handler.